File: src/test/testdata/inspections/unsafeIterators/ObjectIntMap.java
package main;

import com.badlogic.gdx.utils.ObjectIntMap;

import java.util.Iterator;

class ObjectIntMapTest {

  void test() {

    ObjectIntMap<String> ofm = new ObjectIntMap();

    for (ObjectIntMap.Entry<String> e: <warning>ofm</warning>) {
      System.out.println(e.key);
    }

    for (ObjectIntMap.Entry<String> e: new ObjectIntMap.Entries<String>(ofm)) {
      System.out.println(e.key);
    }

    Iterator i1 = <warning>ofm.keys()</warning>;
    Iterator i2 = <warning>ofm.entries()</warning>;
    ObjectIntMap.Values i3 = <warning>ofm.values()</warning>;
    Iterator i4 = <warning>ofm.entries()</warning>;

  }

}

File: src/test/testdata/inspections/unsafeIterators/ObjectIntMap.kt
package main

import com.badlogic.gdx.utils.ObjectIntMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
class ObjectIntMapTest {

  var ofm2: ObjectIntMap<String>? = null

  fun test() {

    val ofm = ObjectIntMap<Any>()


    for (e in <warning>ofm</warning>) {
      println(e.key)
    }

    for (e in ObjectIntMap.Entries<Any>(ofm)) {
      println(e.key)
    }

    val i1 = <warning>ofm.keys()</warning>
    val i2 = <warning>ofm.entries()</warning>
    val i3 = <warning>ofm.values()</warning>
    val i4 = <warning>ofm.entries()</warning>

    <warning>ofm2?.keys()</warning>

    <warning>(ofm2 ?: ObjectIntMap()).values()</warning>

  }

}

